api/src/utils/media/media-utils.ts

```diff
@@ -110,3 +110,28 @@ export const isZipMimetype = (mimetype: string): boolean => {
     regex.test(mimetype)
   );
 };
+
+/**
+ * Returns true if the file is a keyx file, or a zip that contains a keyx file.
+ *
+ * @export
+ * @param {Express.Multer.File} file
+ * @return {*}  {boolean}
+ */
+export function checkFileForKeyx(file: Express.Multer.File): boolean {
+  const mimeType = mime.getType(file.originalname) ?? '';
+  if (file?.originalname.endsWith('.keyx')) {
+    return true;
+  } else if (isZipMimetype(mimeType)) {
+    const zipEntries = parseUnknownZipFile(file.buffer);
+    if (zipEntries.length === 0) {
+      return false;
+    }
+    for (const zipEntry of zipEntries) {
+      if (!zipEntry.fileName.endsWith('.keyx')) {
```

Comment:
If my ZIP contains the following files:
```
[
  'cat.jpg',
  'myfile.keyx',
  'test.csv'
]
```
then this loop would return `false` upon discovering cat.jpg`, even though the documentation says it should "return `true` if the file is [...] a zip that contains a keyx file."

Reply:
Unless there's something I'm missing, I would rewrite this:
```typescript
export const isKeyXFileOrArchive = (file: Express.Multer.File): boolean => {
  // File is a KeyX file if it ends in '.keyx'
  if (file?.originalname.endsWith('.keyx')) {
    return true;
  }

  const mimeType = mime.getType(file.originalname) ?? '';
  if (!isZipMimetype(mimeType)) {
    // File cannot be a KeyX file, since it is not an archive nor does it have a .keyx extension
    return false;
  }

  const zipEntries = parseUnknownZipFile(file.buffer);

  // File is a KeyX archive if some file in the archive has a .keyx extension
  return zipEntries.some((zipEntry) => zipEntry.fileName.endsWith('.keyx');
}
```

Reply:
Yeah good catch, the requirements changed and I forgot to update the description. I cleaned up some of the formatting too.